### PR TITLE
feat(api): Added service detection APIs

### PR DESCRIPTION
**Pull request by didiladi**
This PR builds on the awesome work of @luislira22 in #449.

It adds 4 new APIs:
  * service-detection-api-full-web-request
    for api endpoint /api/config/v1/service/detectionRules/FULL_WEB_REQUEST
  * service-detection-full-web-service
    for api endpoint /api/config/v1/service/detectionRules/FULL_WEB_SERVICE
  * service-detection-opaque-web-request
    for api endpoint /api/config/v1/service/detectionRules/OPAQUE_AND_EXTERNAL_WEB_REQUEST
  * service-detection-opaque-web-service
    for api endpoint /api/config/v1/service/detectionRules/OPAQUE_AND_EXTERNAL_WEB_SERVICE

### CODE REVIEW

**Comment by didiladi** [comment]
Demo configs currently fail with the following error:

```
2021-12-15 15:53:55 ERROR Deployment to environment1 failed with error!
2021-12-15 15:53:55 ERROR 	Failed to update DT object My sample rule_1639580007021513437AllConfigs (HTTP 400)!
    Response was: {"error":{"code":400,"message":"Constraints violated.","constraintViolations":[{"path":"id","message":"ID of rule must match ID in REST call path.","parameterLocation":"PAYLOAD_BODY","location":null}]}}, responsible config: test-resources/integration-all-configs/project/service-detection-opaque-web-service/demorule.json
```

File: cmd/monaco/test-resources/integration-all-configs/project/service-detection-full-web-request/demorule.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      "attributeType": "URL_PATH",
+      "compareOperations": [
+        {
+          "type": "EXISTS",
+          "negate": false
+        }
+      ]
+    }
+  ],
+  "applicationId": null,
```
Comment (UnseenWizzard) [inline cmd/monaco/test-resources/integration-all-configs/project/service-detection-full-web-request/demorule.json:32]: _nitpick:_ 
the `null` values here and in other test configs are a bit odd, especially if someone uses them as a sample.
could we put some dummy values in?

**Review by UnseenWizzard — APPROVED** [review]

File: cmd/monaco/test-resources/integration-all-configs/project/service-detection-full-web-request/demorule.json (hunk 1 above)
Comment (didiladi) [reply cmd/monaco/test-resources/integration-all-configs/project/service-detection-full-web-request/demorule.json]: The `applicationId` is optional in the model. I'll remove it.

**Comment by didiladi** [comment]
> Demo configs currently fail with the following error:
> 
> ```
> 2021-12-15 15:53:55 ERROR Deployment to environment1 failed with error!
> 2021-12-15 15:53:55 ERROR 	Failed to update DT object My sample rule_1639580007021513437AllConfigs (HTTP 400)!
>     Response was: {"error":{"code":400,"message":"Constraints violated.","constraintViolations":[{"path":"id","message":"ID of rule must match ID in REST call path.","parameterLocation":"PAYLOAD_BODY","location":null}]}}, responsible config: test-resources/integration-all-configs/project/service-detection-opaque-web-service/demorule.json
> ```

Fixed this error. The service detection APIs require the body of the `PUT` request to contain the `id` of the rule to change.

**Review by UnseenWizzard — APPROVED** [review]

**Pull request merged**